controllers/reconcilers/configuration/configuration_reconciler.go

```diff
@@ -304,16 +315,47 @@ func (r *Reconciler) Reconcile(ctx context.Context, cr *v1.Observability, s *v1.
 				AccessToken: string(configSecret.Data[RemoteAccessToken]),
 				Channel:     string(configSecret.Data[RemoteChannel]),
 				Tag:         string(configSecret.Data[RemoteTag]),
-				Repository:  repoUrl,
+				Repository:  string(configSecret.Data[RemoteRepository]),
 				Source:      &configSecret,
 			}
+			// if Repository is not a Github repo, attempt to create resources deployment and service
+			if !strings.HasPrefix(repos[configSecret.Name].Repository, "https://api.github.com") && !cr.ExternalSyncDisabled() {
```

Comment:
can we invert that to:
1) if url is a quay url, assume image,
2) otherwise assume file download

The reason is that we might want to be able to download from other locations than Github eventually.

Reply:
We could but there would be other changes needed to support other locations. For example: https://github.com/redhat-developer/observability-operator/blob/main/controllers/reconcilers/configuration/configuration_reconciler.go#L569